### PR TITLE
LightGbm pigstensions

### PR DESCRIPTION
**Pull request by sfilipi**
Ongoing work to address #754 
Those are the LightGbm binary and regression extension methods.

### CODE REVIEW

File: src/Microsoft.ML.Data/Training/TrainerUtils.cs [hunk 1]
```diff
@@ -382,6 +382,16 @@ public static SchemaShape.Column MakeR4ScalarWeightColumn(string weightColumn)
                 return null;
             return new SchemaShape.Column(weightColumn, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false);
         }
+
+        /// <summary>
+        /// Check that the label, feature, weights is not supplied in the args of the constructor.
+        /// Those parameters should be internal if they are not used from the maml help code path.
+        /// </summary>
+        public static void CheckArgsDefaultColNames(IHost env, string defaultColName, string argValue)
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Training/TrainerUtils.cs]: >IHost env [](start = 52, length = 9)

`IExceptionContext` please.

File: src/Microsoft.ML.Data/Training/TrainerUtils.cs [hunk 2]
```diff
@@ -382,6 +382,16 @@ public static SchemaShape.Column MakeR4ScalarWeightColumn(string weightColumn)
...
+        {
+            if (argValue != defaultColName)
+                throw env.Except($"Don't supply a value for the {defaultColName} column in the arguments, as it will be ignored. Specify them in the loader, or constructor instead instead.");
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Training/TrainerUtils.cs]: >instead instead [](start = 172, length = 15)

duplicate duplicate

File: src/Microsoft.ML.Data/Training/TrainerUtils.cs [hunk 3]
```diff
@@ -382,6 +382,16 @@ public static SchemaShape.Column MakeR4ScalarWeightColumn(string weightColumn)
                 return null;
             return new SchemaShape.Column(weightColumn, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false);
         }
+
+        /// <summary>
+        /// Check that the label, feature, weights is not supplied in the args of the constructor.
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Training/TrainerUtils.cs]: >args [](start = 74, length = 4)

Maybe ought to write out `arguments`.

File: src/Microsoft.ML.Data/Training/TrainerUtils.cs [hunk 4]
```diff
@@ -382,6 +382,16 @@ public static SchemaShape.Column MakeR4ScalarWeightColumn(string weightColumn)
...
+        /// Those parameters should be internal if they are not used from the maml help code path.
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Training/TrainerUtils.cs]: >maml help code path [](start = 78, length = 19)

Given the focus we now have I'm a little wary of having language specific to the command line in any new code, that is not somehow directly related to the command line.

**Comment by sfilipi** [comment]
close/reopen to trigger a rebuild.

**Comment by sfilipi** [comment]
close - reopen to trigger rebuild.

**Comment by sfilipi** [comment]
close/reopen to trigger rebuild

**Pull request merged**